SYNC branches

brandonhundt opened the pull request:
Preview is really out of date with Prod.

rpeterman-gp committed:
Merge pull request #235 from PRX/feat/234-add-episode-card-dates

#234: Add dates to episode cards

rpeterman-gp committed:
feat(homepage): add categories links

- fix image layout on small featured story cards

brandonhundt committed:
feat(homepage): new icon and rename header

brandonhundt committed:
feat(homepage): change category icon

rpeterman-gp committed:
feat(homepage): actually use rounded style icon

rpeterman-gp committed:
Merge pull request #237 from PRX/feat/233-homepage-category-links

#233: Add categories links to homepage

rpeterman-gp committed:
Merge pull request #236 from PRX/feat/232-site-header-search-icon

#232: Add search icon button

rpeterman-gp committed:
feat(search): add search page

rpeterman-gp committed:
fix(search): handle missing search info

rpeterman-gp committed:
fix(fetchsearchdata): handle missing queries

rpeterman-gp committed:
debug(api): log search requests and responses

rpeterman-gp committed:
fix(fetchapi): debug: check for req

rpeterman-gp committed:
debug(fetchapi): req object

rpeterman-gp committed:
fix(fetchsearchdata): use api for XHR only

rpeterman-gp committed:
refactor(fetchsearchdata): dynamic load GCS fetch

rpeterman-gp committed:
revert(fetchsearchdata): dynamic load GCS fetch

rpeterman-gp committed:
config(webpack): ignore fetch GCS module

rpeterman-gp committed:
refacor(fetchsearchdata): eval require fetch GCS

rpeterman-gp committed:
fix(fetchsearchdata): require fetch GCS relative

farski committed:
feat(amplify): add Amplify config files

rpeterman-gp committed:
Merge pull request #240 from PRX/feat/amplify-inline-config

feat(amplify): add Amplify config files

brandonhundt committed:
feat(robots.txt): Update sitemap url

brandonhundt committed:
feat(robots.txt): Update sitemap url

brandonhundt committed:
feat(opensearch): revert

rpeterman-gp committed:
feat(robots): change sitemap domain

brandonhundt committed:
Merge pull request #229 from PRX/feat/228-verification-files

#228 robots.txt, ads.txt, opensearch.xml

rpeterman-gp committed:
feat(search): fetch results from lambda URL

rpeterman-gp committed:
feat(opensearch): update domain and search pattern

rpeterman-gp committed:
Merge branch 'production' into feat/238-search-page

rpeterman-gp committed:
feat(appsearch): no dialog wrapper on static page

rpeterman-gp committed:
Merge branch 'feat/238-search-page' of github.com:PRX/theworld.org into feat/238-search-page

rpeterman-gp committed:
fix(app): add app data fetch on init

Makes menus data available to dynamic pages, like search

rpeterman-gp committed:
Merge pull request #243 from PRX/feat/238-search-page

#238: Add search page for OpenSearch

rpeterman-gp committed:
config: update data api domain

rpeterman-gp committed:
feat(episodecard): use nextjs image

rpeterman-gp committed:
feat(sidebarlateststories): update default label

- override label on homepage

rpeterman-gp committed:
Merge pull request #247 from PRX/feat/246-latest-stories-header-update

#246: Update default header label of latest stories sidebar block

rpeterman-gp committed:
fix(storyheader): fix layout of feature header

Header content should not get clipped on shorter window sizes.

brandonhundt committed:
feat(a11y): add aria-labels to CTA close buttons

brandonhundt committed:
Merge pull request #253 from PRX/feat/249-a11y-link-labels

#249 a11y score back to 100

rpeterman-gp committed:
Merge pull request #252 from PRX/fix/251-story-header-title-clipping

#251: Prevent feature header content form being clipped on short screens

rpeterman-gp committed:
fix(stoeyheader): feature header image cover bg

rpeterman-gp committed:
Merge pull request #254 from PRX/fix/story-feature-header-image-cover

HOTFIX: Feature story header should cover background

rpeterman-gp committed:
fix([...alias]): add revalidate to page build

rpeterman-gp committed:
Merge pull request #256 from PRX/fix/add-static-page-revalidate

HOTFIX: Add revalidate to alias page builds

rpeterman-gp committed:
feat(media):add page route

- add app redirects from `file/*` to `media/*`

rpeterman-gp committed:
Merge pull request #258 from PRX/fix/add-audio-route-page

HOTFIX: Add `media/[...slug]` page route

rpeterman-gp committed:
hotfix(media): handle media prefix aliases

brandonhundt committed:
feat(story.body.qa.ts): update styles

rpeterman-gp committed:
config(customhttp): extend cache for next js files

- extend cache for default
- update syntax and capitalization of header names

rpeterman-gp committed:
config(customhttp): Strict-Transport-Security

- max-age 2 years
- preload

rpeterman-gp committed:
config(customhttp): content-security-policy

- remove header

rpeterman-gp committed:
config(customhttp): make default first rule

brandonhundt committed:
feat(sidebarCta_styles): li styling

brandonhundt committed:
Merge pull request #260 from PRX/feat/259-qa-styling

#259 update to styles (QAs, Sidebar CTAs)

rpeterman-gp committed:
config(customhttp): patterns update

- defaul to minimal cache
- refine pattern for build js files
- add rules for api, data, images, and static files

rpeterman-gp committed:
config(customhttp): add subdir rules

rpeterman-gp committed:
Merge pull request #264 from PRX/fix/custom-headers-update-add-cache-times

HOTFIX: custom headers update

rpeterman-gp committed:
feat(search): update search domain

- refactor api endpoint to use fetchApiSearch
- remove uneeded search config

rpeterman-gp committed:
feat(index): add getStaticPaths

rpeterman-gp committed:
fix(insex): remove getStaticPaths

- not allowed on index page

rpeterman-gp committed:
feat(fetchstorydata): always fetch data on server

rpeterman-gp committed:
feat(getstaticprops): use all data props

- refactor homepage fetch to not hit pri api in browser

rpeterman-gp committed:
feat(store): always fetch on server

rpeterman-gp committed:
Merge pull request #265 from PRX/feat/263-content-update-faster

#263: Make content update faster

rpeterman-gp committed:
fix(team): page should render

rpeterman-gp committed:
fix: logic for filling in featured stories arrays

- fix excution params for collection queries

rpeterman-gp committed:
feat: hash page data

rpeterman-gp committed:
fix: convert json to utf8 before base64 encode

rpeterman-gp committed:
fix: switch to md5 hash of data

rpeterman-gp committed:
Merge pull request #267 from PRX/fix/landing-page-featured-story-fill-logic

HOTFIX: Logic for filling in featured stories arrays

rpeterman-gp committed:
feat: hash page data w/ sha256

brandonhundt committed:
fix(404s): static assets

rpeterman-gp committed:
Merge pull request #269 from PRX/fix/isr-hash-data

HOTFIX: Try hashing static props data

brandonhundt committed:
fix(404s): static assets

brandonhundt committed:
Merge pull request #270 from PRX/fix/404-static-assets

HOTFIX: Adds 404ing assets

brandonhundt committed:
feat(readme): triggering a deploy

rpeterman-gp committed:
fix(media): add type and id to page props

rpeterman-gp committed:
Merge pull request #272 from PRX/fix/media-pages-not-rendering-data

HOTFIX: Gets media pages rendering again

rpeterman-gp committed:
fix(story): use `format` for `Story Format` goal

- add `format` prop to full story request params

rpeterman-gp committed:
Merge pull request #274 from PRX/fix/273-plausible-story-format-goal-use-format-prop

#273: Use `format` for `Story Format` Plausible goal

rpeterman-gp committed:
feat: add env variable to control ISR revalidate

- begin process of remapping resource types to store actions directly

rpeterman-gp committed:
fix(storyrelatedlinks): don't render missing image

rpeterman-gp committed:
Merge pull request #275 from PRX/fix/error-related-story-missing-image

HOTFIX: Story related links w/o image no longer cause error

rpeterman-gp committed:
feat(app): only request app data during page init

- refactor page static prop fetches to return data from thunks

rpeterman-gp committed:
Merge pull request #276 from PRX/feat/268-reduce-menu-data-requests

#268: Reduce menu data requests

rpeterman-gp committed:
fix(store): let complete data update existing data

- full image fields shouold include alt
- alias state keys should not start w/ `/`
- update header patterns to not start with '/'
- adjust header cache-control times

rpeterman-gp committed:
config(http): remove max ages from next data path

- update next static path pattern

rpeterman-gp committed:
config(http): rearrange rules

- see if last rule is the one used

rpeterman-gp committed:
config(http): add custom app header to each rule

- Do all rules get used, regardless of pattern?

rpeterman-gp committed:
config: try setting headers in next config

rpeterman-gp committed:
fix(next): api path pattern

rpeterman-gp committed:
config(http): remove cache-control header

rpeterman-gp committed:
debug: node env

rpeterman-gp committed:
feat(api): add cache control headers

- add env variable to control cache control by api endpoint types
- update default cache control header
- remove extra header rules

rpeterman-gp committed:
feat(api): 400  status: cta requests w/ numeric id

rpeterman-gp committed:
feat: add env flag for static page prebuilds

rpeterman-gp committed:
fix(api/cta): add json respons to 400 errors

rpeterman-gp committed:
feat(http): reduce s-maxage to 5 minutes

- defaul to static paths `false`

rpeterman-gp committed:
fix: default static paths to empty array

rpeterman-gp committed:
Merge pull request #277 from PRX/fix/headers-redux-faster-content-updates

HOTFIX: Improve headers and stat reducers for faster content updates

brandonhundt committed:
Merge branch 'preview' into production

rpeterman-gp committed:
fix(story): only use byline items w/ a person

rpeterman-gp committed:
Merge pull request #280 from PRX/fix/story-byline-missing-person

HOTFIX: Only use story byline items that have a person set

rpeterman-gp committed:
fix(htmlcontent): cleanup regex

- cleanup should no long remove closing tags that receed spaces.

rpeterman-gp committed:
feat(config): move social settings to env vars

rpeterman-gp committed:
Merge pull request #281 from PRX/feat/move-fb-access-key-to-env-variable

HOTFIX: Move social platform ID/Tokens to env variables

rpeterman-gp committed:
fix(fetchtermstories): handle `province` field

- add function to generate field names from pathname slug
- add tests for field name generator
- fix cta fetch call on term pages
- handle when term has no stories
- add testing phase to amplify.yml
- various linting fixes

rpeterman-gp committed:
fix(amplify): add phases

rpeterman-gp committed:
config(amplify): move test to first position

rpeterman-gp committed:
config(amplify): add artifacts to test config

brandonhundt committed:
feat(AppCtaBanner): redesign so smaller

rpeterman-gp committed:
config(amplify): pretest make jest dir

brandonhundt committed:
feat(AppCtaBanner): cleanup

rpeterman-gp committed:
Merge pull request #285 from PRX/fix/284-handle-province-term-path

#284: Handle province term paths

rpeterman-gp committed:
feat(assets): GBH logo vector svg

- configure bable to remove `data-name` attributes from svg's

rpeterman-gp committed:
Merge pull request #287 from PRX/fix/ghb-logo-vector-svg

HOTFIX: GBH logo vector svg

brandonhundt committed:
fix(robots): update

brandonhundt committed:
Merge pull request #288 from PRX/fix/update-robots-txt

HOTFIX: update robots.txt file

rpeterman-gp committed:
fix(metatags): sanitize defined values

- add tests for sanitizing metatag values
- update all test describes for consistency

rpeterman-gp committed:
fix(audioheader): only render defined date

rpeterman-gp reviewed (dismissed):
I am not sure how often makes sense to do this. The preview branch is really kept around for major features or refactor that need convenient external review and should be synced up at the onset of that type of work. Not sure if we need to worry about keeping this branch synced at all times.

rpeterman-gp committed:
Merge pull request #290 from PRX/fix/error-missing-metatag-values

HOTFIX: Pages missing expected metatag values should no longer cause 500 errors

brandonhundt committed:
Merge pull request #286 from PRX/fix/283-cls-banner

#283 CTA Banner styling changes for CLS issues

rpeterman-gp committed:
feat(story): refactor story bylines

- add decorator to storyFetchData to combine byline props
- optimize api request props to not include image styles
- fix homepage bug when categories menu is missing
- improve HtmlContent removal of empty html tags

rpeterman-gp committed:
feat(fetchstorydata): backwards compatibility

- make work with old byline prop model prior to admin API update

rpeterman-gp committed:
fix(story): make api request props backwrds compat

rpeterman-gp committed:
fix(bio): render featured stories when missing bio

rpeterman-gp committed:
fix(person): segment audio fetch

- segment audio fetch include segments with multiple authors

rpeterman-gp committed:
Merge pull request #294 from PRX/fix/bio-stories-fetch-pagination

HOTFIX: Render featured stories when missing bio

rpeterman-gp committed:
feat(app): add social share menu

- add bg color to html for better load flash and pulldown of page top

rpeterman-gp committed:
feat(socialsharemenu): touch label contrast

- add backdrop to touch displays

rpeterman-gp committed:
feat(image): add page

- add fetch functions and actions
- add api endpoint

rpeterman-gp committed:
Merge pull request #296 from PRX/feat/289-image-page

#289: Add image page

rpeterman-gp committed:
feat(sharing): switch order of tw and fb links

rpeterman-gp committed:
chore: remove console.log call

rpeterman-gp committed:
Merge pull request #295 from PRX/feat/174-social-share-toolbar

#174: Add social share menu

rpeterman-gp committed:
feat(story): update to use new byline field data

rpeterman-gp committed:
feat(person): query stories w/ new byline field

rpeterman-gp committed:
Merge branch 'production' into feat/refactor-story-bylines

rpeterman-gp committed:
feat(story): use bylines to make plausible events

rpeterman-gp committed:
Merge branch 'feat/refactor-story-bylines' of github.com:PRX/theworld.org into feat/refactor-story-bylines

rpeterman-gp committed:
fix(storyheader): bylines render key

rpeterman-gp committed:
Merge pull request #293 from PRX/feat/refactor-story-bylines

HOTFIX: Refactor story bylines

rpeterman-gp committed:
fix(story): disable lint for no-console

rpeterman-gp committed:
fix(person): story audio queries

- change operator to default, `=`.

rpeterman-gp committed:
Merge pull request #303 from PRX/fix/302-bio-stories-query

#302: Fix person story and audio queries

rpeterman-gp committed:
fix: refactor key hashing to not use base-64

- add `media2.wnyc.org` to next image domains

brandonhundt committed:
feat(apple-touch-icons): add versions

rpeterman-gp committed:
fix(alias): 404 when alias ends w/ file ext

brandonhundt committed:
Merge pull request #304 from PRX/feat/apple-touch-icons

Add missing Apple Touch icons

rpeterman-gp committed:
Merge pull request #306 from PRX/fix/298-404-alias-file-ext

#298: 404 when alias ends w/ file ext

rpeterman-gp committed:
Merge pull request #305 from PRX/fix/301-alias-page-hash-error-non-latin

#301: Refactor key hashing to not use base-64

rpeterman-gp committed:
fix(term): handle term not used on any story

- add story field maps:
  - countries-regions
  - social-tags
  - story-format
- handle false field maps

rpeterman-gp committed:
test(fetchtermstories): add tests for new routes

rpeterman-gp committed:
Merge pull request #307 from PRX/fix/299-missing-tag-routes

#299: Term page routing and rendering fixes

rpeterman-gp committed:
feat(search): add media facet

rpeterman-gp committed:
fix(story): remove `.meta` wrapper from `.info`

rpeterman-gp committed:
Merge pull request #309 from PRX/fix/308-story-byline-not-wrapping

#308: Let bylines wrap

rpeterman-gp committed:
feat(video): add page

rpeterman-gp committed:
feat(appsearch): add media facet results

rpeterman-gp committed:
chore(appsearch): remove console log

rpeterman-gp committed:
fix(search): error when query is cleared

rpeterman-gp committed:
Merge pull request #310 from PRX/feat/297-search-media-facet

#297: Add media facet to search

rpeterman-gp committed:
fix(landingpageheader): layout improvements

- fix issue with content clipping on thinner mobile devices
- improve logo scaling
- improve title heading scaling
- improve small story cards layout on xs mobile

rpeterman-gp committed:
Merge pull request #312 from PRX/fix/311-landing-page-header-clip-content

#311: Landing page header layout improvements

rpeterman-gp committed:
fix(htmlcontent): tweet id matching

- fix error matching tweet id when URL ends with none word character

rpeterman-gp committed:
fix(htmlcontent): tweet embed URL validation

- ensure `twitter.com` domain

rpeterman-gp committed:
Merge pull request #313 from PRX/fix/twitter-embed-url-id-matching

HOTFIX: tweet id matching

rpeterman-gp committed:
feat: move google fonts to local files

rpeterman-gp committed:
Merge pull request #315 from PRX/feat/314-move-google-fonts-local-files

#314: Move Google fonts to local files

rpeterman-gp committed:
fix(api): person strories default range

rpeterman-gp committed:
feat(search): add sort param to api request

rpeterman-gp committed:
Merge pull request #317 from PRX/feat/add-sort-param-to-search

Add sort param to api request

rpeterman-gp committed:
Merge pull request #316 from PRX/fix/bio-page-load-more-stories

HOTFIX: Bio page load more button not loading more stories

rpeterman-gp committed:
fix(story): don't render dates when missing data

rpeterman-gp committed:
Merge pull request #318 from PRX/fix/missing-story-dates

HOTFIX: don't render story dates when missing data

brandonhundt committed:
feat(robots.txt): disallow api path

rpeterman-gp committed:
fix(fetchpriapi): url encode alias before request

- fix audio render error when program is missing
- spellcheck vscode settings

rpeterman-gp committed:
Merge pull request #324 from PRX/fix/322-alias-url-encoding

#322: url encode alias before request

brandonhundt committed:
Merge pull request #319 from PRX/fix/robots-api-path

feat(robots.txt): disallow api path

rpeterman-gp committed:
feat(audioplayer): remove default label text

- spelling correction to IDurationProp interface name

rpeterman-gp committed:
Merge pull request #325 from PRX/feat/323-audio-player-remove-default-text

#323: Remove default label text from audio player

rpeterman-gp committed:
feat: add pubdate metatag

- add to story, episode, and audio pages
- add `metatags-pubdate` to search requests sort param
- move UTC date parsing to lib and add tests
- various spelling corrections

rpeterman-gp committed:
fix(audio): use broadcastDate prop

rpeterman-gp committed:
Merge pull request #326 from PRX/feat/248-add-pubdate-to-metatags

#248: Add `pubdate` metatag to relevant content pages

rpeterman-gp committed:
fix(landingpageheader): sync w/ feature story

- reduce min-heights

rpeterman-gp committed:
Merge pull request #332 from PRX/fix/328-landing-page-header-image-max-height

#328: Landing Page Header Size

brandonhundt committed:
feat(story): remove lede styling

brandonhundt committed:
fix(package-lock.json): remove

rpeterman-gp committed:
fix(fetchcategory): banner image field name

brandonhundt committed:
Merge pull request #335 from PRX/feat/330-remove-lede-styling

#330 remove lede paragraph styling

rpeterman-gp committed:
feat(newsletter): support multiple API keys

- add env var mapping for Critical State client api key

rpeterman-gp committed:
feat(newsletter): debug prop to subscribe err

brandonhundt committed:
feat(episode_cards): remove teasers and other adjustments

brandonhundt committed:
feat(episode_cards): adjust episode cards

rpeterman-gp committed:
Merge pull request #336 from PRX/fix/333-category-header-images-missing

#333: Fix API request for category banner image data

brandonhundt committed:
feat(storyCard): style changes

brandonhundt committed:
feat(storyCard): updated

brandonhundt committed:
feat(storyCard): updated

brandonhundt committed:
feat(storyCardGridStyles): update spacing and styling

brandonhundt committed:
feat(landingPages): adjust spacing

brandonhundt committed:
lint

brandonhundt committed:
lint

brandonhundt committed:
feat(quickLinks): create component and get API resource

brandonhundt committed:
feat(quickLinks): styling

brandonhundt committed:
feat(quickLinks): cleanup

brandonhundt committed:
feat(quickLinks): cleanup

brandonhundt committed:
feat(quickLinks): cleanup

brandonhundt committed:
feat(storyCard): adjust feature styles for mobile

rpeterman-gp committed:
fix(landingpage): easy spacing control

- update spacing method on story card grid

brandonhundt committed:
feat(sidebarEpisode): reworked

brandonhundt committed:
feat(storyCard): try different image proportion

brandonhundt committed:
feat(episodeCard): adjust date and title

brandonhundt committed:
feat(episodeCard): lint

brandonhundt committed:
feat(storyCard): add teasers back

brandonhundt committed:
feat(homepage): move partners block

brandonhundt committed:
feat(storyCard): remove teaser class

brandonhundt committed:
feat(storyCard): clean up styles

brandonhundt committed:
feat(episodeCard): teaser is back

brandonhundt committed:
feat(quickLinksStyles): position breadcrumbs relative

brandonhundt committed:
Merge pull request #340 from PRX/feat/quick-links-menu

#329 Quick Links Menu

brandonhundt committed:
Merge branch 'production' into feat/story-card-updates

brandonhundt committed:
Merge pull request #338 from PRX/feat/episode-card-update

#279 Episode card update

brandonhundt committed:
Merge branch 'production' into feat/story-card-updates

brandonhundt committed:
feat(homepage): clean up sidebar

brandonhundt committed:
feat(homepage): cleaning up from episode card merge

brandonhundt committed:
feat(homepage): boxes, we don't need no stinking boxes

brandonhundt committed:
Merge pull request #339 from PRX/feat/story-card-updates

Story Card Updates

rpeterman-gp committed:
fix(quicklinks): render nothing when no menu

rpeterman-gp committed:
fix(quicklinks): render null when no menu

rpeterman-gp committed:
feat(quicklinks): fetch data w/ homepage data

rpeterman-gp committed:
fix(homepage): parse quick links menu data

rpeterman-gp committed:
Merge pull request #343 from PRX/feat/342-fetch-quicklinks-homepage

#342: fetch data w/ homepage data

rpeterman-gp committed:
fix(enhanceimage): make rel src url's absolute

rpeterman-gp committed:
Merge pull request #345 from PRX/fix/344-htmlcontent-images-relative-urls

#344: Make relative src url's absolute for images in body content

rpeterman-gp committed:
feat(subscribe): debug client env var exists

rpeterman-gp committed:
fix(subscribe): debug client env var as bool

rpeterman-gp committed:
docs(subscribe): file description

rpeterman-gp committed:
fix(story): qa wrap floats

rpeterman-gp committed:
config: update to work with asdf

- upgrade to node 16.16.0
- remove vercel dev dependency

rpeterman-gp committed:
feat: refactor cta data fetching

- move cta fetchers to server
- update cta message parsing to be serializable
- update menu parsing to be serializable
- refactor cta store to group messages by region
- remove content key based cta store
- update page fetchers and components
- update cta components
- update menu components
- various spelling corrections

rpeterman-gp committed:
feat: updates for new cta store model

- add app data fetch to alias page
- caniuse update

rpeterman-gp committed:
feat: add filtering of cta messages

- refactor cta store to include map of resource filter props
- update cta message lookup to include stored filter props
- update cta consumers to include resource type and id

rpeterman-gp committed:
fix(ctaregion): destructure error of shownmessage

rpeterman-gp committed:
fix(getctaregiondata): return null on empty result

rpeterman-gp committed:
fix(media): add app data fetch

rpeterman-gp committed:
Merge pull request #354 from PRX/feat/352-cta-refactor-fetch-filtering

#352: CTA refactor fetch and filtering

rpeterman-gp committed:
Merge branch 'production' into feat/334-newsletter-subscribe-multiple-api-keys

rpeterman-gp committed:
Merge branch 'production' into feat/334-newsletter-subscribe-multiple-api-keys

rpeterman-gp committed:
Merge pull request #337 from PRX/feat/334-newsletter-subscribe-multiple-api-keys

#334: Support multiple Campaign Monitor API keys

rpeterman-gp commented:
This PR is waaaay old and out of date. Going to close to get it out of PR's.
